twitterscraper/tweet.py

```diff
@@ -74,3 +148,25 @@ def from_html(cls, html):
                     pass  # Incomplete info? Discard!
                 except TypeError:
                     pass  # Incomplete info? Discard!
+
+    @staticmethod
+    def extract_links_and_hashtags(soup_html):
+        links = set()
+        hashtags = set()  # 'hashflag' is hashtag + emoji
+
+        for atag in soup_html.find_all('a'):
+            class_ = atag.get('class', None)
+            if 'hashtag' in atag['href']:
+                # atag['href'] like '/hashtag/NBA?src=hash' -> 'NBA'
+                hashtag = atag['href'].split('?')[0].split('/')[-1]
+                hashtags.add(hashtag)
+
+            elif class_:
+                # for links
+                if 'twitter-timeline-link' in class_ and \
+                        'pic.twitter' not in atag.text:  # eliminate picture
+                    expanded_url = atag.get('data-expanded-url', None)
+                    link = expanded_url if expanded_url else atag['href']
```

Comment:
Is there a particular reason not to use `atag['href']` as the default in line 168:
`link = atag.get('data-expanded-url', atag['href'])`
`

Reply:
Good recommendation, thank you!!